src/excalidraw-app/index.tsx

```diff
@@ -290,6 +278,28 @@ const ExcalidrawWrapper = () => {
       resolvablePromise<ImportedDataState | null>();
   }
 
+  const saveDebounced = debounce(
+    async (
+      elements: readonly ExcalidrawElement[],
+      appState: AppState,
+      files: BinaryFiles,
+      onFilesSaved: () => void,
+    ) => {
+      saveToLocalStorage(elements, appState);
+
+      await localFileStorage.saveFiles({
+        elements,
+        files,
+      });
+      onFilesSaved();
+    },
+    SAVE_TO_LOCAL_STORAGE_TIMEOUT,
+  );
+
+  const onBlur = useCallback(() => {
+    saveDebounced.flush();
+  }, [saveDebounced]);
+
```

Comment:
not sure if you moved this inside the component for some reason that's been removed in subsequent commits, but since this isn't needed I'm moving it out again